kolibri/utils/conf.py

```diff
@@ -80,92 +78,126 @@
-            )
-            changed = True
-    if changed:
-        save()
-
-
-def enable_default_plugins():
-    """
-    Enable new plugins that have been added between versions
-    This will have the undesired side effect of reactivating
-    default plugins that have been explicitly disabled by a user.
-    However, until we add disabled plugins to a blacklist, this is
-    unavoidable.
-    """
-    global config
-    changed = False
-    for module_path in DEFAULT_PLUGINS:
-        if module_path not in config["INSTALLED_APPS"]:
-            config["INSTALLED_APPS"].append(module_path)
-            logger.warning(
-                (
-                    "Default plugin {mod} not found in configuration. To re-disable it, run:\n"
-                    "   $ kolibri plugin {mod} disable"
-                ).format(mod=module_path)
-            )
-            changed = True
-
-    if changed:
-        save()
+# These values are encoded on the config dict as sets
+# so they need to be treated specially for serialization
+# and deserialization to/from JSON
+SET_KEYS = ["INSTALLED_APPS", "DISABLED_APPS"]
+
+
+class ConfigDict(dict):
+    def __init__(self):
+        # If the settings file does not exist or does not contain
+        # valid JSON then create it
+        self.update(
+            {
+                #: Everything in this list is added to django.conf.settings.INSTALLED_APPS
+                # except disabled ones below
+                "INSTALLED_APPS": DEFAULT_PLUGINS,
+                #: Everything in this list is removed from the list above
+                "DISABLED_APPS": [],
+            }
+        )
+        if os.path.isfile(conf_file):
+            try:
+                # Open up the config file and load settings
+                # use default OS encoding
+                with open(conf_file, "r") as kolibri_conf_file:
+                    self.update(json.load(kolibri_conf_file))
+                return
+            except json.JSONDecodeError:
+                pass
+        logger.info("Initialize kolibri_settings.json..")
+        self.save()
+
+    @property
+    def ACTIVE_PLUGINS(self):
+        return list(self["INSTALLED_APPS"] - self["DISABLED_APPS"])
+
+    def update(self, new_values):
+        """
+        Updates current configuration with ``new_values``. Does not save to file.
+        """
+        values_copy = new_values.copy()
+        for key in SET_KEYS:
+            if key in values_copy:
+                values_copy[key] = set(values_copy[key])
+        super(ConfigDict, self).update(values_copy)
+
+    def save(self):
+        # use default OS encoding
+        config_copy = self.copy()
+        for key in SET_KEYS:
+            if key in config_copy:
+                config_copy[key] = list(config_copy[key])
+        with open(conf_file, "w") as kolibri_conf_file:
+            json.dump(config_copy, kolibri_conf_file, indent=2, sort_keys=True)
+
+    def autoremove_unavailable_plugins(self):
```

Comment:
it's not clear to me why we would actually want to do this. Seems like it would be better to leave the plugin enabled, but simply warn that it couldn't be loaded?